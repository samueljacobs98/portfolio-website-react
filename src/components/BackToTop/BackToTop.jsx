import React from "react";
import toTop from "../../assets/svgs/to-top.svg";

import "./BackToTop.scss";

const BackToTop = () => {
  return (
    <>
      <a href="#home" className="mobile-back-to-top">
        <img src={toTop} alt="back-to-top" />
      </a>
      <a href="#home" className="back-to-top">
        Back to top
      </a>
    </>
  );
};

export default BackToTop;
