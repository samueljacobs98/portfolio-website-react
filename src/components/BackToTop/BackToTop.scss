@use "../../assets/scss/variables" as *;

.back-to-top {
  display: none;
  color: $colour-grey;
  font-size: 1rem;
  position: fixed;
  left: 10%;
  bottom: 1.2rem;
}

.mobile-back-to-top {
  position: fixed;
  right: 1.2rem;
  bottom: 1.2rem;
}

@media only screen and (min-width: 768px) {
  .mobile-back-to-top {
    display: none;
  }

  .back-to-top {
    display: block;
  }
}
