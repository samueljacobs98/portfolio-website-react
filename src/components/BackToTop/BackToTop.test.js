import { render } from "@testing-library/react";
import BackToTop from "./BackToTop";

test("should match snapshot", () => {
  const { container } = render(<BackToTop />);
  expect(container).toMatchSnapshot();
});
