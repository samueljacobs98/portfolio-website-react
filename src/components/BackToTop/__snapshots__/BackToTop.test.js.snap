// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <a
    class="mobile-back-to-top"
    href="#home"
  >
    <img
      alt="back-to-top"
      src="to-top.svg"
    />
  </a>
  <a
    class="back-to-top"
    href="#home"
  >
    Back to top
  </a>
</div>
`;
